src/Components/ArtistFilter/Artworks.tsx

```diff
@@ -14,41 +9,91 @@ import { Artworks_filtered_artworks } from "../../__generated__/Artworks_filtere
 
 interface Props {
   filtered_artworks: Artworks_filtered_artworks
-  relay: RelayPaginationProp
+  relay: RelayRefetchProp
   artistID: string
 }
 
 const PAGE_SIZE = 10
 
 class Artworks extends React.Component<Props> {
-  loadMoreArtworks(filters) {
+  loadPage(filters, page) {
+    this.props.relay.refetch(
+      {
+        page,
+        size: PAGE_SIZE,
+        filteredArtworksNodeID: this.props.filtered_artworks.__id,
+      },
+      null,
+      error => {
+        if (error) {
+          console.error(error)
+        }
+        filters.setPage(page)
+      }
+    )
+  }
+
+  loadNextPage(filters) {
     const hasMore = this.props.filtered_artworks.artworks.pageInfo.hasNextPage
-    const endCursor = this.props.filtered_artworks.artworks.pageInfo.endCursor
+
     if (hasMore) {
-      // TODO: Should refetchConnection keep appending records?
-      this.props.relay.refetchConnection(
-        PAGE_SIZE,
-        error => {
-          if (error) {
-            console.error(error)
-          }
-          filters.incrementPage()
-        },
-        { cursor: endCursor }
+      return this.loadPage(filters, filters.state.page + 1)
+    }
+  }
+
+  loadPrevPage(filters) {
+    if (filters.state.page > 1) {
+      return this.loadPage(filters, filters.state.page - 1)
+    }
+  }
+
+  renderPaginationBar(filters) {
+    let page
+    const pages = []
+    for (
+      page = 1;
+      page < this.props.filtered_artworks.artworks.totalPages + 1;
+      page++
+    ) {
+      pages.push(
+        <span
+          data-page={page}
```

Comment:
Oh yea, we can definitely tackle this when we make this a more proper component, but I couldn't figure out how to get the click handler to work on `page`, b/c of references all the click handlers just point to the last page (probably something simple, can figure out later when we take another pass).

Reply:
npnp, we can clean up in another pass